### PR TITLE
Fixes hostile simple animals not attacking barricades and sandbags

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
bugfix: You can no longer permanently block hostile simple mobs with sandbags and other barricades.
/:cl:

rip legion strategy rip kor's promise 

```
<KorMobile> which is a bug
that I will fix one day
```

### CODE REVIEW

**Comment by Bawhoppen** [comment]
they shouldnt be able to instantly destroy it though

**Comment by ChangelingRain** [comment]
It's not instant, silly.

**Comment by Man-Shroom-** [comment]
then what the fuck is the sandbag even for

**Comment by RemieRichards** [comment]
Make the objects a typecache on the mob, so that things can pick and choose what they consider "destroy-able surroundings"

**Comment by ChangelingRain** [comment]
I'm not here to fix this horrible line, I'm just here to fix that one bug and then pretend this line doesn't exist

**Comment by RemieRichards** [comment]
@ChangelingRain No, you are changing what is considered a valid target for destroy-able surroundings, so make the change.

**Comment by ChangelingRain** [comment]
I'm pretending this line doesn't exist I'm pretending I'm not fixing it

**Comment by RemieRichards** [comment]
Yep that's what I wanted, :+1:

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -40,6 +50,7 @@
 	..()
 	if(!targets_from)
 		targets_from = src
+	environment_target_typecache = typecacheof(environment_target_typecache)
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: why not just make this a static variable so it only needs to be set up once

Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: So you can have subtypes that attack certain things but not other things.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: My initial POINT for this change being made was to allow subtypes to choose to ignore sandbags, which before this change, would have been a forced target for all hostile mobs that break surroundings, Hostile mobs are supposed to be customisable, not locked to hardcoded setups.

the fact that typecaches are a thing and speed up that line was just a happy accident to this change.

**Pull request merged**